### PR TITLE
Updated read_excel docstring to match style guide formatting

### PR DESCRIPTION
**Pull request by GarrettDaniel**
Fixes `read_excel()` docstring to match docstring formatting style guide:
- Issue: https://github.com/pandas-dev/pandas/issues/53674. 
- Style Guide: https://pandas.pydata.org/docs/development/contributing_docstring.html

### CODE REVIEW

File: pandas/io/excel/_base.py [hunk 1]
```diff
@@ -79,7 +79,7 @@
     )
 _read_excel_doc = (
     """
-Read an Excel file into a pandas DataFrame.
+Read an Excel file into a ``pandas`` ``DataFrame``.
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:82]: I'm thinking we shouldn't wrap `DataFrame` throughout. In my opinion, it doesn't need the highlighting, and it can serve as a distraction.

File: pandas/io/excel/_base.py [hunk 2]
```diff
@@ -101,61 +101,61 @@
...
     Engine compatibility :
 
-    - "xlrd" supports old-style Excel files (.xls).
-    - "openpyxl" supports newer Excel file formats.
-    - "odf" supports OpenDocument file formats (.odf, .ods, .odt).
-    - "pyxlsb" supports Binary Excel files.
+    - ``"xlrd"`` supports old-style Excel files (.xls).
+    - ``"openpyxl"`` supports newer Excel file formats.
+    - ``"odf"`` supports OpenDocument file formats (.odf, .ods, .odt).
+    - ``"pyxlsb"`` supports Binary Excel files.
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:158]: I would suggest no double quotes here (so just e.g. ` ``xlrd`` `). We can just be referring to the engine itself, rather than the string argument.

File: pandas/io/excel/_base.py [hunk 3]
```diff
@@ -181,70 +181,70 @@
...
+    * If ``keep_default_na=False``, and ``na_values`` are not specified, no
+      strings will be parsed as ``NaN``.
+
+    Note that if ``na_filter=False``, the ``keep_default_na`` and
+    ``na_values`` parameters will be ignored.
 na_filter : bool, default True
-    Detect missing value markers (empty strings and the value of na_values). In
-    data without any NAs, passing na_filter=False can improve the performance
+    Detect missing value markers (empty strings and the value of ``na_values``). In
+    data without any NAs, ``passing na_filter=False`` can improve the performance
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:218]: passing shouldn't be included (just `na_filter=False`)

File: pandas/io/excel/_base.py [hunk 4]
```diff
@@ -279,13 +279,13 @@
 
     .. versionadded:: 1.2.0
 
-dtype_backend : {{"numpy_nullable", "pyarrow"}}, defaults to NumPy backed DataFrames
-    Which dtype_backend to use, e.g. whether a DataFrame should have NumPy
-    arrays, nullable dtypes are used for all dtypes that have a nullable
-    implementation when "numpy_nullable" is set, pyarrow is used for all
-    dtypes if "pyarrow" is set.
+dtype_backend : {{"numpy_nullable", "pyarrow"}}, defaults to ``numpy`` backed ``DataFrames``
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:282]: NumPy is the correct capitalization, now?

File: pandas/io/excel/_base.py [hunk 5]
```diff
@@ -356,7 +356,7 @@
 1       NaN      2
 2  #Comment      3
 
-Comment lines in the excel input file can be skipped using the `comment` kwarg
+Comment lines in the excel input file can be skipped using the ``comment`` ``kwarg``
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:359]: kwarg here should not be included. Can you write out the full phrase here: keyword argument.

File: pandas/io/excel/_base.py [hunk 6]
```diff
@@ -181,70 +181,70 @@
...
-      result 'foo'
+    * ``dict``, e.g. ``{'foo' : [1, 3]}`` -> parse columns 1, 3 as date and call
+      result ``'foo'``
 
     If a column or index contains an unparsable date, the entire column or
     index will be returned unaltered as an object data type. If you don`t want to
-    parse some cells as date just change their type in Excel to "Text".
-    For non-standard datetime parsing, use ``pd.to_datetime`` after ``pd.read_excel``.
+    parse some cells as date, just change their type in Excel to "Text".
+    For non-standard ``datetime`` parsing, use ``pd.to_datetime`` after ``pd.read_excel``.
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:236]: I don't think datetime here is being used in a technical sense (e.g. specifying a package or snippet of code), and so shouldn't highlighted.

File: pandas/io/excel/_base.py [hunk 7]
```diff
@@ -345,7 +345,7 @@
 1   string2    2.0
 2  #Comment    3.0
 
-True, False, and NA values, and thousands separators have defaults,
+``True``, ``False``, ``NaN`` values, and thousands of separators have defaults,
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:348]: The docs use NA as opposed to NaN in various places because it can refer to pd.NA. Can you revert this change.

File: pandas/io/excel/_base.py [hunk 8]
```diff
@@ -101,61 +101,61 @@
...
       (0-indexed).
-    * If list of string, then indicates list of column names to be parsed.
+    * If list of ``str``, then indicates list of column names to be parsed.
     * If callable, then evaluate each column name against it and parse the
       column if the callable returns ``True``.
 
     Returns a subset of the columns according to behavior above.
 dtype : Type name or dict of column -> type, default None
-    Data type for data or columns. E.g. {{'a': np.float64, 'b': np.int32}}
+    Data type for data or columns. E.g. ``{'a': np.float64, 'b': np.int32}``
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:146]: I believe you need to leave the double curly braces `{{` and `}}` for jinja-style templating.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR! Overall looks good - some requests.

File: pandas/io/excel/_base.py (hunk 1 above)
Comment (GarrettDaniel) [reply pandas/io/excel/_base.py:82]: Good call.  I wasn't sure on that one so I'm glad you provided some clarity on it.  It could go either way, but I agree that it gets distracting with how often DataFrame is referenced in the docs.

File: pandas/io/excel/_base.py (hunk 2 above)
Comment (GarrettDaniel) [reply pandas/io/excel/_base.py:158]: Good call!  I'll make sure to change that in the commit I'm working on.

File: pandas/io/excel/_base.py (hunk 3 above)
Comment (GarrettDaniel) [reply pandas/io/excel/_base.py:218]: Nice catch!

File: pandas/io/excel/_base.py (hunk 4 above)
Comment (GarrettDaniel) [reply pandas/io/excel/_base.py:282]: It is, I just wasn't sure if it would be better to go with NumPy or ``numpy``, but like the comment above with DataFrame and pandas, I agree that it would be distracting to make every instance of NumPy a code block.

File: pandas/io/excel/_base.py (hunk 6 above)
Comment (GarrettDaniel) [reply pandas/io/excel/_base.py:236]: Good point.  If it was referencing the ``datetime`` package that would be a more appropriate time to put it in a codeblock.

File: pandas/io/excel/_base.py (hunk 7 above)
Comment (GarrettDaniel) [reply pandas/io/excel/_base.py:348]: On it.

File: pandas/io/excel/_base.py (hunk 8 above)
Comment (GarrettDaniel) [reply pandas/io/excel/_base.py:146]: I do agree that we need that for jinja-style templating, but from what I can tell, we aren't using jinja-style templating in this case.  If we were, wouldn't we be passing in parameters to be read in and rendered like a formatted string? (https://realpython.com/primer-on-jinja-templating/).  At the moment, it renders like a regular string, but this seems like a perfect use case for a code block.

image

File: pandas/io/excel/_base.py (hunk 5 above)
Comment (GarrettDaniel) [reply pandas/io/excel/_base.py:359]: Good call.  Should ``comment`` still be in a code block in this case since it's referencing the name of a parameter?

Comment (rhshadrach) [reply pandas/io/excel/_base.py:359]: Yes - I think so.

File: pandas/io/excel/_base.py (hunk 8 above)
Comment (GarrettDaniel) [reply pandas/io/excel/_base.py:146]: Ok that's good to know.  In that case, should we just leave it to render as a string with the double curly braces so that CI won't fail?  I suppose we could put that in a code block but it might look a little strange (i.e. ```{'a': np.float64, 'b': np.int32}}``)

Comment (rhshadrach) [reply pandas/io/excel/_base.py:146]: > (i.e. `{{'a': np.float64, 'b': np.int32}}`)

When you do this, do both curly braces render in the docs? I would expect only one renders.

Comment (GarrettDaniel) [reply pandas/io/excel/_base.py:146]: I've been trying to get the docs to render as per the guide (https://pandas.pydata.org/docs/development/contributing_documentation.html#building-the-documentation), but I keep running into `ModuleNotFoundError: No module named 'pandas.__libs.pandas_parser'` when running `python3 make.py html`.  This is most likely because of the version of pandas I have on my laptop, but I tried upgrading to the newest version and uninstalling + reinstalling and the error still persists.  At this point, I'll commit it with the code block around the braces and if the CI build still fails I'll just remove the code block and leave it to render as standard text.

Comment (rhshadrach) [reply pandas/io/excel/_base.py]: Did you create a development environment and compile/install pandas?

https://pandas.pydata.org/docs/development/contributing_environment.html

Comment (GarrettDaniel) [reply pandas/io/excel/_base.py]: I've tried a few of the different methods on that doc, namely with Docker, DevContainers, and Mamba, and none of them were able to successfully `import pandas`.  I believe it has something to do with a corporate proxy or security issue, but I either run into that same `ModuleNotFoundError`, or I run into `ERROR: Disabling PEP 517 processing is invalid: project specifies a build backend of mesonpy in pyproject.toml` when trying to run `python -m pip install -e . --no-build-isolation --no-use-pep517`.  If you have any ideas on this, let me know.

Comment (rhshadrach) [reply pandas/io/excel/_base.py]: I believe you're going off the stable (e.g. 2.0.3) version of the docs for building pandas. When reading docs on development, it's best to read the dev docs as we will break things well before releasing :laughing:. See here:

https://pandas.pydata.org/pandas-docs/dev/development/contributing_environment.html#step-3-build-and-install-pandas

Comment (rhshadrach) [reply pandas/io/excel/_base.py]: > Otherwise, I've made all of the other requested changes in the other comments.

The CI is failing; this will need to be addressed.

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.
